File: Math/2961.Double Modular Exponentiation/Solution.java
class Solution {
    public List<Integer> getGoodIndices(int[][] variables, int target) {
        List<Integer> ans = new ArrayList<>();
        for (int i = 0; i < variables.length; i++) {
            int a = variables[i][0], b = variables[i][1];
            int c = variables[i][2], m = variables[i][3];
            if (qpow(qpow(a, b, 10), c, m) == target) {
                ans.add(i);
            }
        }
        return ans;
    }

    public int qpow(int a, int b, int mod) {
        int ans = 1;
        while (b > 0) {
            if ((b & 1) == 1) {
                ans = ans * a % mod;
            }
            a = a * a % mod;
            b >>= 1;
        }
        return ans;
    }
}

File: Math/2961.Double Modular Exponentiation/Solution.py
class Solution:
    def getGoodIndices(self, variables: List[List[int]], target: int) -> List[int]:
        ans = []
        for i, v in enumerate(variables):
            a, b, c, m = v
            if (a ** b % 10) ** c % m == target:
                ans.append(i)
        return ans